core/src/main/scala/org/apache/gearpump/jarstore/JarStoreService.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gearpump.jarstore
+
+import java.util.ServiceLoader
+
+import akka.actor.Props
+import java.io.File
+import com.typesafe.config.Config
+import org.apache.gearpump.util.Constants
+import scala.collection.JavaConverters._
+
+case class FilePath(url: String)
+
+trait JarStoreService {
+  /**
+    * Whether the given url match the scheme.
+   */
+  def matchScheme(path: String): Boolean
+
+  /**
+    * Creates a JarStore based on the provided path.
+   * This function will be called from the server side.
+   * @param rootPath The JarStore root path
+   * @return
+   */
+  def props(rootPath: String): Props
+
+  /**
+    * This function will copy the local file to the remote JarStore, called from client side.
+   * @param localFile The local file
+   * @param remotePath The path on JarStore
+   */
+  def copyFromLocal(localFile: File, remotePath: FilePath)
+
+  /**
+    * This function will copy the remote file to local file system, called from client side.
+   * @param localFile The destination of file path
+   * @param remotePath The remote file path from JarStore
+   */
+  def copyToLocalFile(localFile: File, remotePath: FilePath)
+}
+
+object JarStoreService {
+  lazy val jarstoreServices: List[JarStoreService] = {
+    ServiceLoader.load(classOf[JarStoreService]).asScala.toList
+  }
+
+  def getJarStoreService(rootPath: String): JarStoreService = {
+    jarstoreServices.find(_.matchScheme(rootPath)).get
```

Comment:
but it will be difficult to find the cause without looking at the source code if we don't provide any help info